hal/inc/deviceid_hal.h

```diff
@@ -83,6 +83,11 @@ int hal_get_device_serial_number(char* str, size_t size, void* reserved);
 // TODO: Move this function to an appropriate module
 int hal_get_device_secret(char* data, size_t size, void* reserved);
 
+/**
+ * Get the device's hardware version.
+ */
+int hal_get_device_hw_version();
```

Comment:
Let's instead use `int hal_get_device_hw_revision(uint32_t* revision, void* reserved)` and return `system_error_t` instead.

Reply:
Fixed